pkg/api/api.go

```diff
@@ -540,6 +540,9 @@ func (hs *HTTPServer) registerRoutes() {
 		// metrics
 		// DataSource w/ expressions
 		apiRoute.Post("/ds/query", requestmeta.SetSLOGroup(requestmeta.SLOGroupHighSlow), authorize(ac.EvalPermission(datasources.ActionQuery)), hs.getDSQueryEndpoint())
+		if hs.Features.IsEnabledGlobally(featuremgmt.FlagDsQueryConvert) {
```

Comment:
this one seems weird and confusing -- like `/ds/query` is is a payload that could accept mixed resuls with no apiversion?  

I would rather not put effort into this pattern and instead shift towards the explicitly versioned apis for each service.

Reply:
I want to be able to test / use this without having to enable all endpoints, let's achieve one thing at a time. If by the time we are fine testing the feature versioned apis are stable and enabled by default, this endpoint won't be used and we can remove the feature flag and the endpoint altogether.

Reply:
if the end result is an unimplemented, 404/500 whatever I don't get the difference -- using/testing this will likely only add confusion.

Reply:
? not sure I follow. The end result is the same regardless of the endpoint used (apis or api). It shouldn't add confusion for the end user (plugin dev) using the feature since it should be transparent for them which endpoint is being used. It will be abstracted away by the frontend.

The difference is that enabling `datasourceAPIServers` will change the query endpoing AND add the convert endpoint, while enabling `dsQueryConvert` will only add the convert endpoint. The blast radious is way smaller and therefore safer to start just with the second.

Reply:
/api/ds/query-convert does not include the requested apiVersion -- IIUC correctly it only includes the saved apiVersion.  Yes this is maybe all theoretical now given that there is just one, but I am concernd that we hide that and believe it can/should be meaningful without the version.  If you goal is that you can call something you can always call:

`/apis/{group}/{apiVersion}/..../convert`

so IIUC, the same result, but at least scoped to a place that we want in the end

Reply:
I know, as you say, since we are still in v0, theoretically, having it or not is not meaningful. By the time we have a v1, definetly this "legacy" endpoint won't be used. 

> your goal is that you can call something you can always call:
> /apis/{group}/{apiVersion}/..../convert

Yes, that's [what I am doing in my tests](https://github.com/grafana/grafana-plugin-examples/pull/340/commits/67f91f6a22088d08cfa16a4a9a6f44f8c657c02a).

Again, this is just to be able to test / use query conversion _without_ having to route query requests to the new endpoint but if you are strongly against this I can remove this code.

Reply:
This is confusing 😄 Partly, because it's a new feature not supported earlier for plugins and hence no existing `/api/ds/query/convert`, and partly only exposing this for /apis wouldn't have any use for the foreseeable future. Maybe since no use now we (I at least) have a hard time seeing the path/steps towards this being actually used 🤷 

To move this forward, lets skip the `/api/ds/query/convert` for now and decide on the next steps?